streambed/src/crypto.rs

```diff
@@ -60,15 +66,42 @@ pub fn encrypt(bytes: &mut [u8], key: &[u8], salt: &[u8]) {
     }
 }
 
+/// Sign an array of bytes using a key. Given the same bytes and key, this will always
+/// result in the same output.
+pub fn sign(bytes: &[u8], key: &[u8]) -> Vec<u8> {
+    use hmac::Mac;
+    let mut mac = HmacSha256::new_from_slice(key).unwrap();
+    mac.update(bytes);
+    mac.finalize().into_bytes().to_vec()
+}
+
+/// Verify that a signed value is equal to an original array of bytes
+pub fn verify(bytes: &[u8], key: &[u8], compare_bytes: &[u8]) -> bool {
+    use hmac::Mac;
+    let mut mac = HmacSha256::new_from_slice(key).unwrap();
+    mac.update(bytes);
+    mac.verify_slice(compare_bytes).is_ok()
+}
+
+/// Hash an array of bytes with a salt. Yields a SHA-512 hash.
+pub fn hash(bytes: &[u8], salt: &[u8; SALT_SIZE]) -> Vec<u8> {
+    let mut hasher = Sha512::new();
+    hasher.update(salt);
+    hasher.update(bytes);
+    let mut hash = salt.to_vec();
+    hash.extend(hasher.finalize());
+    hash
+}
+
 #[cfg(test)]
 mod tests {
     use super::*;
 
     #[test]
-    fn encrypt_decrypt() {
+    fn test_encrypt_decrypt() {
         let bytes = b"The quick brown fox jumps over the lazy dog.";
-        let key = [0_u8; 16];
-        let salt = [0_u8; 12];
+        let key = [0_u8; KEY_SIZE];
+        let salt = [0_u8; SALT_SIZE];
```

Comment:
This test seems to fail if the key and salt are not all zeroes

Reply:
L189’a value would change…

Reply:
Not sure what you mean sorry :)

Reply:
Ah, line 109 - you're doing two asserts, right